packages/core/src/helpers/selector.ts

```diff
@@ -197,17 +197,20 @@ export function scopeNestedSelector(
                 },
                 { ignoreList: [`selector`] }
             );
-            const parentRef = first.type === `nesting`;
-            const globalSelector = first.type === `pseudo_class` && first.value === `global`;
-            const startWithScoping = rootScopeLevel
-                ? scopeAst.nodes.every((node, i) => {
-                      return matchTypeAndValue(node, outputAst.nodes[i]);
-                  })
-                : false;
-            let nestedMixRoot = false;
+            // merge scope flags
+            const nestStartWithNesting = first.type === `nesting`;
+            const nestedStartWithGlobal =
+                rootScopeLevel && first.type === `pseudo_class` && first.value === `global`;
```

Comment:
the change is here - the addition of `rootScopeLevel &&` to limit the behavior of not prepending the parent scope for root mixins